toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -609,6 +653,26 @@ func (g *PkgGraph) AllNodesFrom(rootNode *PkgNode) []*PkgNode {
 	return nodes
 }
 
+func (g *PkgGraph) printLookupTable() {
+	for pkgname, list := range g.lookupTable() {
+		logger.Log.Debugf("----------------------------------")
+		logger.Log.Debugf("Package name: %s", pkgname)
+		for _, node := range list {
+			//check if node.RunNode is nil
+			if node.RunNode == nil {
```

Comment:
I think `RunNode` is never expected to be `nil` - that would be a bug. Unless this change makes that OK.

Reply:
I agree that `RunNode` cannot be `nil`. **It won't be `nil`** with my change. It is just a sanity check before trying to print properties inside node.RunNode, so there won't be any null dereference.

Reply:
Btw. do we even call this function? I think we may be able to remove it.

Reply:
This function is not called currently. I used it to get the state of the `lookupTable` as I was modifying it. I found it very useful for debugging. It'd be good to have in the code.

Reply:
This function is not called anywhere in the toolkit. It was added for debugging purposes. I removed it now. As part of improving debuggability of toolkit, we can add it later.